cli/main.go

```diff
@@ -69,16 +69,31 @@ func main() {
 		return
 	}
 
-	client := client.Client{
-		URL:   "https://" + CONFIG.Domain,
-		Token: CONFIG.Token,
-	}
+	var client clientPkg.Client
 
 	app := cli.NewApp()
 	app.Name = "draupnir"
 	app.Version = version.Version
 	app.Usage = "A client for draupnir"
-	cli.AppHelpTemplate = fmt.Sprintf("%s%s", cli.AppHelpTemplate, quickStart)
+	app.CustomAppHelpTemplate = fmt.Sprintf("%s%s", cli.AppHelpTemplate, quickStart)
+	app.Flags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "insecure",
+			Usage: "don't validate certificates when connecting to draupnir",
+		},
+	}
+
+	// Construct a client before we run any of our commands
+	app.Before = func(c *cli.Context) error {
+		client = clientPkg.NewClient(
+			fmt.Sprintf("https://%s", CONFIG.Domain),
```

Comment:
Doesn't this need to be http for `--insecure`?

Reply:
Nope, as the goal is to target the https endpoint but not check certs

Reply:
Cool, given that's the ultimate aim this makes sense.